error TS2344: Type 'ws' does not satisfy the constraint 'abstract new (...args: any) => any'

### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [x] This is an issue with the Node library

### Describe the bug

Typechecking code that imports openai websocket client fails with the following error:
```
node_modules/openai/beta/realtime/websocket.d.ts:7:19 - error TS2344: Type 'ws' does not satisfy the constraint 'abstract new (...args: any) => any'.

7 }) ? InstanceType<ws> : any;
                    ~~

  node_modules/openai/beta/realtime/websocket.d.ts:6:22
    6     WebSocket: infer ws;
                           ~~
    This type parameter might need an `extends abstract new (...args: any) => any` constraint.
```

### To Reproduce

[repro link](https://stackblitz.com/edit/openai-ts-issue?file=src%2Fmain.ts)

### Code snippets

```JavaScript

```

### OS

any

### Node version

Node v23

### Library version

openai v4.84.0

--- a/src/beta/realtime/websocket.ts
+++ b/src/beta/realtime/websocket.ts
@@ -11,7 +11,7 @@ interface MessageEvent {
 type _WebSocket =
   typeof globalThis extends (
     {
-      WebSocket: infer ws;
+      WebSocket: infer ws extends abstract new (...args: any) => any;
     }
   ) ?
     // @ts-ignore

